src/SFA.DAS.EmployerApprenticeshipsService.Application/Queries/GetTransactionsDownloadResultViewModel/GetTransactionsDownloadQueryHandler.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.EAS.Application.Validation;
+using SFA.DAS.EAS.Domain.Data.Repositories;
+using SFA.DAS.EAS.Domain.Models.UserProfile;
+using SFA.DAS.HashingService;
+
+namespace SFA.DAS.EAS.Application.Queries.GetTransactionsDownloadResultViewModel
+{
+    public class GetTransactionsDownloadQueryHandler : IAsyncRequestHandler<GetTransactionsDownloadQuery, GetTransactionsDownloadResponse>
+    {
+        private readonly CurrentUser _currentUser;
+        private readonly ITransactionRepository _transactionRepository;
+        private readonly ITransactionFormatterFactory _transactionsFormatterFactory;
+        private readonly IMembershipRepository _membershipRepository;
+        private readonly IHashingService _hashingService;
+
+        public GetTransactionsDownloadQueryHandler(
+            CurrentUser currentUser,
+            ITransactionRepository transactionRepository,
+            ITransactionFormatterFactory transactionsFormatterFactory, 
+            IMembershipRepository membershipRepository,
+            IHashingService hashingService)
+        {
+            _currentUser = currentUser;
+            _transactionRepository = transactionRepository;
+            _transactionsFormatterFactory = transactionsFormatterFactory;
+            _membershipRepository = membershipRepository;
+            _hashingService = hashingService;
+        }
+
+        public async Task<GetTransactionsDownloadResponse> Handle(GetTransactionsDownloadQuery message)
+        {
+            var membership = await _membershipRepository.GetCaller(message.AccountHashedId, _currentUser.ExternalUserId);
+
+            if (membership == null)
+            {
+                throw new UnauthorizedAccessException();
+            }
+
+            var validationResult = new ValidationResult();
+
+            if (message.StartDate.DateInFuture)
```

Comment:
The upper bound is validated but the lower bound isn't, is that ok? We could also validate that `StartDate` isn't greater than `EndDate`.